htlcswitch/switch.go

```diff
@@ -2689,3 +2720,99 @@ func (s *Switch) evaluateDustThreshold(link ChannelLink,
 	// If we reached this point, this HTLC is fine to forward.
 	return false
 }
+
+// failAliasUpdate prepares a ChannelUpdate for a failed incoming or outgoing
+// HTLC on an option-scid-alias channel. If the associated channel is not an
+// option-scid-alias channel, this function will return nil and the caller is
+// expected to handle this gracefully. In this case, a return to the original
+// non-alias behavior is expected.
+func (s *Switch) failAliasUpdate(scid lnwire.ShortChannelID,
+	incoming bool) *lnwire.ChannelUpdate {
+
+	// This function does not defer the unlocking because of the database
+	// lookups for ChannelUpdate.
+	s.indexMtx.RLock()
+
+	if IsAlias(scid) {
+		// The alias SCID was used. In the incoming case this means
+		// the channel is zero-conf as the link sets the scid. In the
+		// outgoing case, the sender set the scid to use and may be
+		// either the alias or the confirmed one, if it exists.
+		realScid, ok := s.aliasToReal[scid]
+		if !ok {
+			s.indexMtx.RUnlock()
+
+			// The real, confirmed SCID does not exist yet. Use the
+			// alias for lookup. No SCID replacement is necessary.
+			update, err := s.cfg.FetchLastChannelUpdate(scid)
+			if err != nil {
+				return nil
+			}
+
+			return update
+		}
+
+		s.indexMtx.RUnlock()
+
+		// Fetch the SCID via the confirmed SCID and replace it with
+		// the alias.
+		update, err := s.cfg.FetchLastChannelUpdate(realScid)
+		if err != nil {
+			return nil
+		}
+
+		// In the incoming case, we want to ensure that we don't leak
+		// the UTXO in case the channel is private. In the outgoing
+		// case, since the alias was used, we do the same thing.
+		update.ShortChannelID = scid
+		sig, err := s.cfg.SignAliasUpdate(update)
```

Comment:
Ah ok, I see the usage now...

Do we need to generate a fresh signature for this each time though?